packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -403,6 +426,23 @@ - (BOOL)setCaptureSessionPreset:(FLTResolutionPreset)resolutionPreset withError:
   return YES;
 }
 
+/// Finds the highest available resolution in terms of pixel count for the given device
+- (AVCaptureDeviceFormat *)getHighestResolutionFormatFor:(AVCaptureDevice*)captureDevice {
+    AVCaptureDeviceFormat *bestFormat = nil;
+    NSUInteger maxPixelCount = 0;
+    for ( AVCaptureDeviceFormat *format in [_captureDevice formats] ) {
+        CMVideoDimensions res = CMVideoFormatDescriptionGetDimensions(format.formatDescription);
+        NSUInteger height = res.height;
+        NSUInteger width = res.width;
+        NSUInteger pixelCount = height * width;
+        if ( pixelCount > maxPixelCount ) {
+          maxPixelCount = pixelCount;
+          bestFormat = format;
+        }
+    }
+    return bestFormat;
```

Comment:
what if `[_captureDevice formats]` is empty? do we want to handle nil case?

Reply:
We do handle it actually, this method returns nil if there is no bestFormat and switch case from setCaptureSessionPreset doesn't break and falls through to other cases